etna/analysis/outliers/prediction_interval_outliers.py

```diff
@@ -78,18 +104,19 @@ def get_anomalies_prediction_interval(
     else:
         ts_inner = create_ts_by_column(ts, in_column)
     outliers_per_segment = {}
-    time_points = np.array(ts.index.values)
     model_instance = model(**model_params)
     model_instance.fit(ts_inner)
     lower_p, upper_p = [(1 - interval_width) / 2, (1 + interval_width) / 2]
-    prediction_interval = model_instance.predict(
-        deepcopy(ts_inner), prediction_interval=True, quantiles=[lower_p, upper_p]
-    )
     for segment in ts_inner.segments:
-        predicted_segment_slice = prediction_interval[:, segment, :][segment]
-        actual_segment_slice = ts_inner[:, segment, :][segment]
+        ts_segment = select_segments_subset(ts=deepcopy(ts_inner), segments=[segment])
+        ts_segment.df = ts_segment.df.dropna()
+        prediction_interval = model_instance.predict(
+            deepcopy(ts_segment), prediction_interval=True, quantiles=[lower_p, upper_p]
```

Comment:
Do we need this `deepcopy`?

Reply:
Yes, without `deepcopy` ts_segment will be changed by `model_instance.predict`

Reply:
Ok, understood.